shared/cosmos_db.py

```diff
@@ -92,19 +92,53 @@ def get_conversation_data(conversation_id):
         logging.info(
             f"[CosmosDB] customer sent an inexistent conversation_id, saving new {conversation_id}"
         )
-        conversation = container.create_item(body={"id": conversation_id})
+        # Initialize with default structure including all required keys
+        conversation = container.create_item(
+            body={
+                "id": conversation_id,
+                "conversation_data": {
+                    "start_date": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                    "history": [],  # Initialize empty history list
+                    "memory_data": "",
+                    "interaction": {},
+                }
+            }
+        )
 
-    # get conversation data
+    # Get conversation data with complete default structure
     conversation_data = conversation.get(
         "conversation_data",
         {
             "start_date": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
-            "history": [],
+            "history": [],  # Ensure history exists in default structure
             "memory_data": "",
             "interaction": {},
         },
     )
 
+    # Double-check that history exists
```

Comment:
consider updating this comment to: `# Double-check that history exists and initialize it if missing`